src/libstore/globals.cc

```diff
@@ -114,6 +114,11 @@ void loadConfFile()
        ~/.nix/nix.conf or the command line. */
     globalConfig.resetOverridden();
 
+    /* Alway consider this option as overridden so environment
+       variables inherited from current environment can be sent to
+       daemon. */
+    settings.environment.overridden = true;
+
```

Comment:
This is a bit irky, and also somewhat wrong if the client config doesn't include the daemon config (for instance when building on a remote machine): Even if the client config doesn't specify the `environment` setting, its default value will be sent and override whatever the daemon might have set.

Do we really need the “get from the environment if not specified” logic when parsing the `environment` configuration option? I'd rather stay simple and predictable (at least for a first version) and just have this variable be a list of key-value pairs.